content/sips/sip-187.md

```diff
@@ -159,13 +161,96 @@ function _updateCachedSynthDebtsWithRates(
     }
 ```
 
+#### Updating debt cache for minting and burning sUSD
+
+When minting and burning sUSD, the `cachedSum[sUSD]` and `_cachedDebt` values that tracks the amount of sUSD should be updated to reflect the increase or decrease in the sUSD supply and the cached debt also updated to track the amount of sUSD that is minted / burned. It would be more accruate to update the amount of sUSD issued or burned, instead of updating the cached value of sUSD by reading the sUSD total supply as this could include non-SNX collateral issued sUSD such as Multi-collateral loans and shorts between snapshots. The excluded debts will be updated during the snapshot instead.
+
+**DebtCache Contract**
+```
+contract DebtCache is BaseDebtCache {
+    function increaseCachedsUSDDebt(
+        uint amount
+    ) external onlyIssuer {
+        _cachedSynthDebt[sUSD] = _cachedSynthDebt[sUSD].add(amount);
+        _cachedDebt = _cachedDebt.add(amount);
+    }
+
+    function decreaseCachedsUSDDebt(
+        uint amount
+    ) external onlyIssuer {
+        _cachedSynthDebt[sUSD] = _cachedSynthDebt[sUSD].sub(amount);
+        _cachedDebt = _cachedDebt.add(amount);
+    }
+}
+```
+
+**Issuer Contract**
+
+```
+Contract Issuer {
+    function _issueSynths(
+        address from,
+        uint amount,
+        bool issueMax
+    ) internal {
+        (uint maxIssuable, uint existingDebt, uint totalSystemDebt, bool anyRateIsInvalid) = _remainingIssuableSynths(from);
+        _requireRatesNotInvalid(anyRateIsInvalid);
+
+        if (!issueMax) {
+            require(amount <= maxIssuable, "Amount too large");
+        } else {
+            amount = maxIssuable;
+        }
 
+        // Keep track of the debt they're about to create
+        _addToDebtRegister(from, amount, existingDebt, totalSystemDebt);
 
+        // record issue timestamp
+        _setLastIssueEvent(from);
+
+        // Create their synths
+        synths[sUSD].issue(from, amount);
+
+        // Account for the issued debt in the cache
+        debtCache().increaseCachedsUSDDebt(amount);
```

Comment:
[np] i would replace all this with just this line stating that it's the diff (and the current line that's being replaced). Overall this feels way too technical for the SIP.